### DISCUSSION TITLE
Database Schematic

### DISCUSSION THREAD

**Discussion by johnceci-m3** [postalserver/postal#2465] [Help with using Postal]
Does anyone have a good database schematic of how to the messages are stored in the DB?  How the tables relate to each other?  This would be useful for doing some digging into the data that is stored but I can't seem to find anything in the docs.  I am starting down the rabbit hole, but if there is already something out there, it would be helpful.

**Comment by sinnbeck** [comment]
Why not just connect to the database with something like dbeaver? You can then easily view the schematic

Quick example from the main database (each server has its own database as well)
![image](https://github.com/postalserver/postal/assets/13980140/7898793f-4801-487f-9d3b-21cab083b9db)

**Comment by johnceci-m3** [comment]
I could, but some explanation more so than just using PK/FK/Constraints would be useful for those of us that might want to dig a big deeper....

**Reply by sinnbeck** [reply]
Fair enough :) Sadly I havent dug much into them myself
